defi/src/storeTvlInterval/computeTVL.ts

```diff
@@ -1,98 +1,110 @@
+      return;
+    }
+
+    const normalisedKey = key.startsWith("0x")
+      ? `ethereum:${key.toLowerCase()}`
+      : !key.includes(":")
+      ? `coingecko:${key.toLowerCase()}`
+      : key;
+
+    // sols case sensitive so no normalising
+    if (key == normalisedKey || key.startsWith("solana:")) return;
+
+    balances[normalisedKey] =
+      normalisedKey in balances
+        ? (Number(balances[normalisedKey]) + Number(balances[key])).toString()
+        : balances[key];
+
+    delete balances[key];
+  });
+
   const eth = balances[ethereumAddress];
   if (eth !== undefined) {
     balances[weth] = new BigNumber(balances[weth] ?? 0).plus(eth).toFixed(0);
     delete balances[ethereumAddress];
   }
-  const PKsToTokens = {} as { [t: string]: string[] };
-  const readKeys = Object.keys(balances)
-    .map((address) => {
-      let prefix = "";
-      if(address.startsWith("0x")){
-        prefix = "ethereum:"
-      } else if(!address.includes(":")){
-        prefix = "coingecko:"
-      }
-      let normalizedAddress = address.toLowerCase()
-      if(address.startsWith("solana:")){
-        normalizedAddress = address
-      }
-      const PK = `${prefix}${normalizedAddress}`;
-      if (PKsToTokens[PK] === undefined) {
-        PKsToTokens[PK] = [address];
-        return PK;
-      } else {
-        PKsToTokens[PK].push(address);
-        return undefined;
-      }
-    })
-    .filter((item) => item !== undefined) as string[];
+
+  return balances;
+}
+
+async function fetchTokenData(
+  timestamp: "now" | number,
+  balances: { [address: string]: string }
+) {
   const readRequests = [];
-  for (let i = 0; i < readKeys.length; i += 100) {
-    const body = {
-      "coins": readKeys.slice(i, i + 100),
-    } as any
-    if(timestamp !== "now"){
-      body.timestamp = timestamp;
-    }
+  const burl = "https://coins.llama.fi/prices/";
+  const historical =
+    timestamp == "now" ? "current/" : `historical/${timestamp}/`;
+
+  for (let i = 0; i < Object.keys(balances).length; i += step) {
+    const coins = Object.keys(balances)
+      .slice(i, i + step)
+      .reduce((p, c) => p + `${c},`, "");
     readRequests.push(
-      fetch("https://coins.llama.fi/prices", {
-        method: "POST",
-        body: JSON.stringify(body)
-      }).then((r) => r.json()).then(r=>{
-        return Object.entries(r.coins).map(
-        ([PK, value])=>({
-          ...(value as any),
-          PK
-        })
+      fetch(`${burl}${historical}${coins}`, { method: "GET" }).then((r) =>
+        r.json()
       )
-    })
     );
   }
-  const tokenData = ([] as any[]).concat(...(await Promise.all(readRequests)));
+
+  const responses: any[] = await Promise.all(readRequests);
+  return responses.map((g) => g.coins);
+}
+
+const confidenceThreshold = 0.5;
+const step = 40;
+export default async function (
+  balances: { [address: string]: string },
+  timestamp: "now" | number,
+  staleCoins: any
+) {
+  staleCoins; // can get rid of stalecoins type?
```

Comment:
why is this here lol, doesnt this do nothing?

Reply:
yeah I just put it there so it'd run